docs/default.nix

```
@@ -6,73 +6,160 @@
+        && !lib.hasPrefix "${inputs.self}/modules/" path;
     };
 
-  nixos = makeOptionsDoc (
-    lib.nixosSystem {
-      inherit (pkgs) system;
-      modules = [
-        inputs.home-manager.nixosModules.home-manager
-        inputs.self.nixosModules.stylix
-        ./settings.nix
-      ];
-    }
-  );
-
-  homeManager = makeOptionsDoc (
-    inputs.home-manager.lib.homeManagerConfiguration {
-      inherit pkgs;
-      modules = [
-        inputs.self.homeManagerModules.stylix
-        ./settings.nix
-        {
-          home = {
-            homeDirectory = "/home/book";
-            stateVersion = "22.11";
-            username = "book";
-          };
-        }
-      ];
-    }
-  );
+  # Returns an attribute set of module names and their corresponding option
+  # documentation.
+  makeModuleOptionsDoc =
+    configuration:
+    lib.mapAttrs (
+      module: _:
+      makeOptionsDoc {
+        inherit configuration;
+        pathFilter = lib.hasPrefix "${inputs.self}/modules/${module}/";
+      }
+    ) (builtins.readDir "${inputs.self}/modules");
+
+  nixosModuleOptionsDoc = makeModuleOptionsDoc nixosConfiguration;
+  homeManagerModuleOptionsDoc = makeModuleOptionsDoc homeManagerConfiguration;
+
+  modulePageScript = lib.pipe "${inputs.self}/modules" [
+    builtins.readDir
+    (lib.mapAttrsToList (
+      module: _: ''
+        writeModulePage \
+          ${module} \
+          ${homeManagerModuleOptionsDoc.${module}} \
+          ${nixosModuleOptionsDoc.${module}}
+      ''
+    ))
+    lib.concatStrings
+  ];
 
 in
 pkgs.stdenvNoCC.mkDerivation {
   name = "stylix-book";
   src = ./.;
+  buildInputs = with pkgs; [
+    mdbook
+    mdbook-alerts
+  ];
 
   patchPhase = ''
+    # The generated documentation has headings at level 2, but we want level 3
+    # so they can be nested under the sections for each module system.
+    REDUCE_HEADINGS='s/^## /### /'
+
+    # The "declared by" links point to a file which only exists when the docs
+    # are built locally. This removes the links.
+    REMOVE_DECLARED_BY='/*Declared by:*/,/^$/d'
+
+    function writeOptions() {
+      platformName="$1"
+      optionsFile="$2"
+      outputFile="$3"
+
+      printf '\n## %s options\n' "$platformName" >>"$outputFile"
+
+      if [[ -s "$optionsFile" ]]; then
+        sed \
+          --expression "$REDUCE_HEADINGS" \
+          --expression "$REMOVE_DECLARED_BY" \
+          <"$optionsFile" \
+          >>"$outputFile"
+      else
+        printf '*%s.*\n' "None provided." >>"$outputFile"
+      fi
+    }
+
+    function writeModulePage() {
+      moduleName="$1"
+      homeManagerOptionsFile="$2"
+      nixosOptionsFile="$3"
+
+      readmeFile="${inputs.self}/modules/$moduleName/README.md"
+      page="options/modules/$moduleName.md"
+      outputFile="src/$page"
+
+      if [ -f "$outputFile" ]; then
+        echo "Please move docs/src/options/modules/$moduleName.md to modules/$moduleName/README.md" >&2
```

Consider the following adjustments:

```suggestion
        printf \
          '%s should not be used. Move it to %s\n' \
          "docs/src/options/modules/$moduleName.md" \
          "modules/$moduleName/README.md" \
          >&2
```